### DISCUSSION TITLE
Best Approach to Implementing a User Class?

### DISCUSSION THREAD

**Discussion by MiguelGibo** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#169] [Q&A]
Hi, this is a sister post to #170, I wanted to ask y'all's input on what the core attributes and methods for our User class should be.

## Suggested Implementation
**Attributes:**
* Username (str): Unique identifier used by system to distinguish between students.
* Display name (str): Name that appears on profile and during search.
* email (str): used for sign-up/log-in and account recovery.
* password (str): Temporarily stored as plaintext for initial development purposes, but encryption should be implemented as a priority task by a small dev group to ensure security.
* Role (str): Designates if a user is an admin, directive_member, department_member, or regular student. This will determine permission levels, but role-based permissions will be fully implemented in a future iteration. For now, users may pick any role, but a task should be assigned to a small team to define how roles and permissions should function.
* responsible_rooms (list(str)): List of room IDs that the user is responsible for managing or overseeing. This would mainly apply to admins or department members, allowing them to manage room availability and/or reservations.

**Methods:**
* make_event(event_name (str), room_id (str), day (int), time (int)): Creates an object of type event class (suggested implementation for event class will be made in separate discussion post). Allows a user to reserve a room to create an event. Leave as placeholder until event class is implemented. Returns a success or failure message.
* rsvp_event(event_id (str)): Adds user to list of attendees to an event. Adds event to user's reservations list.
* view_reservations(): lets users see their reservations list.
* update_profile(new_username (str), new_display_name (str), new_email (str), new_password (str), new_role (str)): Allows user to change their user information. **Note: That new_role aspect should be left non-functional for now and will be completed when role permissions are fully implemented in a future task, and that username should always be unique.**

Best,
Miguel :)

**Comment by pierreg-vales** [comment]
Hi! Great idea! One attribute should be a map that holds all the tags the user selcts. 

Create an issue to create the class body and attributes
